### PR TITLE
Moved THREE.Geometry out of core.

### PR DESCRIPTION
**Pull request by mrdoob**
Moved `Geometry` to `examples/jsm/deprecated`.
Moved `DirectGeometry` inside of `Geometry` and added `toBufferGeometry()` in `Geometry`.

### CODE REVIEW

**Comment by Mugen87** [comment]
`BufferGeometry.setFromObject()` and `BufferGeometry.updateFromObject()` also rely on `Geometry` and have to be removed.

**Comment by mrdoob** [comment]
The main thing left in this PR are the modifiers and I think the sanest thing to do is to convert them to use `BufferGeometry`...

Won't be able to work on that until Monday though.

**Comment by Mugen87** [comment]
Um, not sure how easy that is. You probably want to use convert calls to `BufferGeometry().fromGeometry();` to `Geoemtry.toBufferGeometry()` for now.

**Comment by Mugen87** [comment]
During my tests I have noticed one issue:

There is now `Geometry.toBufferGeometry()` but it only works with mesh geometries. We actually need the logic from `BufferGeometry.setFromObject()` so also lines and point geometries do work.

File: examples/jsm/deprecated/Geometry.js [hunk 1]
```diff
@@ -1365,6 +1366,120 @@ Geometry.prototype = Object.assign( Object.create( EventDispatcher.prototype ),
 
 	},
 
+	toBufferGeometry: function () {
+
+		const geometry = new DirectGeometry().fromGeometry( this );
+
+		const positions = new Float32Array( geometry.vertices.length * 3 );
+		this.setAttribute( 'position', new BufferAttribute( positions, 3 ).copyVector3sArray( geometry.vertices ) );
```
Comment (Mugen87) [inline examples/jsm/deprecated/Geometry.js:1374]: `this` can be used here. It should be:
```
const bufferGeometry = new BufferGeometry();

const positions = new Float32Array( geometry.vertices.length * 3 );
bufferGeometry.setAttribute( 'position', new BufferAttribute( positions, 3 ).copyVector3sArray( geometry.vertices ) );
```

Comment (mrdoob) [reply examples/jsm/deprecated/Geometry.js:1374]: Yep! This code is half-done.

File: src/Three.d.ts [hunk 2]
```diff
@@ -32,7 +32,6 @@ export * from './textures/CubeTexture';
 export * from './textures/CanvasTexture';
 export * from './textures/DepthTexture';
 export * from './textures/Texture';
-export * from './geometries/Geometries';
```
Comment (Mugen87) [inline src/Three.d.ts:35]: It seems this line is required so the geometry generators are exported.

Comment (mrdoob) [reply src/Three.d.ts:35]: Yeah, not sure what happened there. Will fix.

File: src/renderers/webgl/WebGLGeometries.js [hunk 3]
```diff
@@ -63,15 +62,9 @@ function WebGLGeometries( gl, attributes, info, bindingStates ) {
...
-
-			if ( geometry._bufferGeometry === undefined ) {
-
-				geometry._bufferGeometry = new BufferGeometry().setFromObject( object );
-
-			}
+		} else {
 
-			buffergeometry = geometry._bufferGeometry;
+			console.log( 'TODO: Remove this', geometry );
```
Comment (Mugen87) [inline src/renderers/webgl/WebGLGeometries.js:67]: When this is merged, I'd like to refactor `WebGLGeometries` slightly. The class can be implemented now a bit simpler.

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
@Mugen87 🙌🙌🙌

**Comment by Mugen87** [comment]
This is exciting! We've worked so long so it's possible to do this step! Glad to see we've made it! 🙌

**Comment by Mugen87** [comment]
FYI: I've setup a forum topic to inform the community about this change:

https://discourse.threejs.org/t/three-geometry-will-be-removed-from-core-with-r125/22401

**Comment by GGmemeda** [comment]
Geometry deprecation ，i want to get vertices, faces, faceVertexUvs from bufferGeometry,Is there any other good way?

**Comment by Mugen87** [comment]
@GGmemeda Please use the [forum](https://discourse.threejs.org/) for help requests. Similar questions like the following one were already answered there:

https://discourse.threejs.org/t/facevertexuvs-for-buffergeometry/23040

**Comment by mjurczyk** [comment]
@mrdoob I think that's not possible to change wiki with a PR, so a little note here:
https://github.com/mrdoob/three.js/wiki/JSON-Object-Scene-format-4#example-of-scene-with-raw-geometry-light-and-a-material-with-multiple-uv-texture-assignment
Still refers to the original Geometry at the bottom of the doc (mentioned by user 선과 on discord - https://discord.com/channels/685241246557667386/686304808994340865/806395575482056704)

Same applies to:
- https://github.com/mrdoob/three.js/wiki/JSON-Model-format-3#core
- https://github.com/mrdoob/three.js/wiki/Getting-Started (technically it works, BoxGeometry is overridden with BoxBufferGeometry now, but for clarity it could probably use -BufferGeometry 🤔 ?)

**Comment by Mugen87** [comment]
Thanks for the feedback. I've clean up the Wiki pages.

**Pull request merged**